src/modules/ChannelList/dux/__tests__/reducers.spec.js

```diff
@@ -906,4 +919,51 @@ describe('Channels-Reducers', () => {
       expect(hideWithPreventAutoOnPreventAutoState.allChannels.length).toEqual(preventAutoUnhideParamsState.allChannels.length + 1);
     });
   });
+  describe.only('isChannelJustCreated', () => {
+    const prevState = { ...mockData, channelListQuery: { includeEmpty: false } };
+
+    it('should not add a newly created channel to the ChannelList if it is just created and the current user is the only member', () => {
+      const newChannel = {
+        ...creatingChannel,
+        members: [user1],
+        createdAt: new Date(),
+        invitedAt: new Date(),
+        lastMessage: null,
+      };
+      (isChannelJustCreated).mockReturnValue(true);
+
+      const nextState = reducers(prevState, {
+        type: actionTypes.ON_USER_JOINED,
+        payload: newChannel,
+      });
+
+      expect(isChannelJustCreated).toHaveBeenCalledWith(newChannel);
+      expect(nextState.allChannels.find(channel => channel.url === newChannel.url)).toBeUndefined();
+      expect(nextState.currentChannel).toEqual(mockData.currentChannel);
+      expect(nextState).toEqual(prevState); // Check if nextState and prevState are deeply equal
+    });
+
+    it('should add a newly created channel to the ChannelList if it is not just created', () => {
+      const newChannel = {
+        ...creatingChannel,
+        members: [user1, user2],
+        createdAt: 123456789,
+        invitedAt: 129999967,
+        lastMessage: null,
+      };
+      (isChannelJustCreated).mockReturnValue(false);
+      (getNextChannel).mockReturnValue(newChannel);
```

Comment:
doesn't need to wrap the variables with `()`?

Reply:
Oh right. My bad

Reply:
Just changed the way of mocking functions https://github.com/sendbird/sendbird-uikit-react/pull/1124/commits/d691222b5b7c6e7f4cd78767cceb1a13579c972f#diff-9536f4716bc6385ea2e3e35374ac6abb3ec4259a4881777212f637e53f1fce71R912